exercises/lamar_moore/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+namespace Shapes
+{
+	/** List of possible shape types.
+	*/
+	typedef enum
```

Comment:
Not a bad solution here. However, 

1. If we are to add more shapes then we'll have to add to this enum. If we're writing a library, this header and enum will be read-only to our clients. So they may not be able to update the definition when they subtype Shape.
1. In a large project (such as Mantid), adding to a header which is used throughout the project could lead to a seriously long recompilation. 
1. This isn't quite as bad as making Shape aware of it's derived types, but it's close. 

For those reasons, I would have chosen to have the type as a std::string.